compiler/rustc_middle/src/ty/fast_reject.rs

```diff
@@ -312,6 +312,7 @@ impl DeepRejectCtxt {
             // Impls cannot contain these types as these cannot be named directly.
             ty::FnDef(..) | ty::Closure(..) | ty::Generator(..) => false,
 
+            // Placeholder types don't unify with anything on their own
             ty::Placeholder(..) | ty::Bound(..) => false,
```

Comment:
weird that we allow `ty::Bound` but not `ConstKind::Bound` :shrug: both should be unreachable (or well, reachable with ty/const binders)